File: src/com/draekko/adbtools/action/QuickListAction.java
package com.draekko.adbtools.action;

import com.intellij.ide.actions.QuickSwitchSchemeAction;
import com.intellij.openapi.actionSystem.*;
import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class QuickListAction extends QuickSwitchSchemeAction implements DumbAware {

    protected void fillActions(@Nullable final Project project,
                               @NotNull final DefaultActionGroup group,
                               @NotNull final DataContext dataContext) {

        if (project == null) {
            return;
        }

        addAction("adbtools.action.UninstallAction", group);
        addAction("adbtools.action.KillAction", group);
        addAction("adbtools.action.StartAction", group);
        addAction("adbtools.action.StartWithDebuggerAction", group);
        addAction("adbtools.action.RestartAction", group);
        addAction("adbtools.action.RestartWithDebuggerAction", group);
        addAction("adbtools.action.ClearDataAction", group);
        addAction("adbtools.action.ClearDataAndRestartAction", group);
        addAction("adbtools.action.ClearDataAndRestartWithDebugerAction", group);
    }

    protected boolean isEnabled() {
        return true;
    }

    private void addAction(final String actionId, final DefaultActionGroup toGroup) {
        final AnAction action = ActionManager.getInstance().getAction(actionId);

        // add action to group if it is available
        if (action != null) {
            toGroup.add(action);
        }
    }

    protected String getPopupTitle(AnActionEvent e) {
        return "ADB Operations Popup";
    }

}

File: src/main/kotlin/io/github/raghavsatyadev/adbidea/action/QuickListAction.kt
package io.github.raghavsatyadev.adbidea.action

import com.intellij.ide.actions.QuickSwitchSchemeAction
import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project

class QuickListAction : QuickSwitchSchemeAction(), DumbAware {
  override fun fillActions(project: Project?, group: DefaultActionGroup, dataContext: DataContext) {

    if (project == null) {
      return
    }

    addAction("io.github.raghavsatyadev.adbidea.action.UninstallAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.KillAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.StartAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.RestartAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.ClearDataAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.ClearDataAndRestartAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.RevokePermissionsAction", group)
    group.addSeparator()
    addAction("io.github.raghavsatyadev.adbidea.action.StartWithDebuggerAction", group)
    addAction("io.github.raghavsatyadev.adbidea.action.RestartWithDebuggerAction", group)
  }

  private fun addAction(actionId: String, toGroup: DefaultActionGroup) {
    // add action to group if it is available
    ActionManager.getInstance().getAction(actionId)?.let { toGroup.add(it) }
  }

  override fun isEnabled() = true

  override fun getPopupTitle(e: AnActionEvent) = "ADB Operations Popup"
}
